google_guest_agent/accounts_unix.go

```diff
@@ -27,21 +27,30 @@ import (
 	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
 )
 
-func getUID(path string) string {
+func getUIDAndGID(path string) (string, string) {
 	if dir, err := os.Stat(path); err == nil {
 		if stat, ok := dir.Sys().(*syscall.Stat_t); ok {
-			return fmt.Sprintf("%d", stat.Uid)
+			return fmt.Sprintf("%d", stat.Uid), fmt.Sprintf("%d", stat.Gid)
 		}
 	}
-	return ""
+	return "", ""
 }
 
-func createUser(ctx context.Context, username, uid string) error {
+func createUser(ctx context.Context, username, uid, gid string) error {
 	config := cfg.Get()
 	useradd := config.Accounts.UserAddCmd
 	if uid != "" {
 		useradd = fmt.Sprintf("%s -u %s", useradd, uid)
 	}
+	if gid != "" {
+		groupadd := config.Accounts.GroupAddCmd
+		groupadd = fmt.Sprintf("%s -g %s", groupadd, gid)
+		cmd, args := createUserGroupCmd(groupadd, "", username)
+		if err := run.Quiet(ctx, cmd, args...); err != nil {
```

Comment:
will there be a case where group already exists? as probably this will fail if group already exists? I believe we want to continue in that case and not return error.

Reply:
I don't think there would be a case where the group with name `username` already exists because this block of code will only run if the user does not exist on the system. 

However, the `groupadd` command could return an error if the GID already exists. In that case, we would want to return the error since this indicates that some other group already claimed the GID so we can't create the user with that GID.